### PR TITLE
Ai003 preprocess text - done

### PR DESCRIPTION
**Pull request by Andrewwango**
As an AI Engineer I need the data prepared so it’s ready to be fed into a model 

For the following Acceptance Criteria see `ai/02_train_model.ipynb`. This ticket treats training the word embedding model to get vectors for words. In the next ticket ("train model") I will write the scoring algorithm to get bias predictions from word embedding vectors.

- Remove null values 
- Vectorize 
- Create a function to stem text with chosen stemmer 
- Use some stopwords (carefully curate)

### CODE REVIEW

**Review by KewinTaoAnh — APPROVED** [review]

File: ai/02_train_model.ipynb [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5e6097e1-68ac-4870-a7ce-5ecab0c8b832",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class TrainingCorpus:\n",
+    "    def __iter__(self):\n",
+    "        corpus_path = datapath('/Users/andrew.wang/Documents/academy/project/kainosRecruitmentApi-TeamA/ai/data/corpus/dataset_wikibios_merged.txt')\n",
```
Comment (ldonnellykainoscom) [inline ai/02_train_model.ipynb:91]: You should be able to get rid of the datapath and make the path relative. It worked for me.

File: ai/02_train_model.ipynb [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    "        for line in open(corpus_path):\n",
+    "            tokens = utils.simple_preprocess(line)\n",
```
Comment (ldonnellykainoscom) [inline ai/02_train_model.ipynb:93]: Looks like there's an option here to remove accents from words. That might be a good from of preprocessing to include: https://tedboy.github.io/nlps/generated/generated/gensim.utils.simple_preprocess.html

File: ai/02_train_model.ipynb [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9c13d6b9-a782-4da9-a5bc-6dab88a22bdf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from gensim.models import Word2Vec"
```
Comment (ldonnellykainoscom) [inline ai/02_train_model.ipynb:136]: Good to keep imports at the top just like in .py files

File: ai/02_train_model.ipynb [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8096a6f2-a70a-4dc6-a289-e90a71f112fb",
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "model = Word2Vec(sentences=corpus, workers=4)"
```
Comment (ldonnellykainoscom) [inline ai/02_train_model.ipynb:167]: Just a thought: was wondering if there's a nice way to show progress when training? I can only see logging lines

File: ai/02_train_model.ipynb (hunk 2 above)
Comment (Andrewwango) [reply ai/02_train_model.ipynb:93]: Saw that, I chose not to do that because I don't want random French/Turkish words etc. turning into English words when their accents are removed, e.g. [Eyüp, Istanbul](https://en.wikipedia.org/wiki/Ey%C3%BCp) turning into [eyup](https://en.wiktionary.org/wiki/ey_up)

File: ai/02_train_model.ipynb (hunk 4 above)
Comment (Andrewwango) [reply ai/02_train_model.ipynb:167]: Yeah..... not really. I tried to do that weeks ago when doing the doc2vec experiments (which wraps word2vec model in gensim) and it's quite tricky, requires creating a callback to output progress on epoch end, as there is no single training loop iterator in the code (multithreading wraps it). Even more difficult to print training loss, see [here](https://stackoverflow.com/questions/54888490/gensim-word2vec-print-log-loss). But because word2vec is only a few epochs, the logging is good enough I think.

**Comment by Andrewwango** [comment]
Cheers @ldonnellykainoscom , re-requested review

File: ai/02_train_model.ipynb (hunk 4 above)
Comment (ldonnellykainoscom) [reply ai/02_train_model.ipynb:157]: Fair enough. Not worth the effort

**Review by ldonnellykainoscom — APPROVED** [review]
Good to go!

**Review by hellofadude — APPROVED** [review]
